File: R/0001.R
#' Problem 1
#'
#' If we list all the natural numbers below 10 that are multiples of 3
#' or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
#'
#' Find the sum of all the multiples of 3 or 5 below 1000.

library(magrittr)
start <- Sys.time()

ans <-
  1:999 %>%
  .[. %% 3 == 0 | . %% 5 == 0] %>%
  sum()

end <- Sys.time()
paste('The answer to Problem 1 is:', ans,
      '\n<< Returned in', end - start, 'seconds >>') %>%
  cat()

File: julia/0001.jl
# Problem 1
#
# If we list all the natural numbers below 10 that are multiples of 3
# or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
#
# Find the sum of all the multiples of 3 or 5 below 1000.

start = Dates.now()

ans = sum(filter(x -> (x % 3 == 0 || x % 5 == 0), 1:999))

finish = Dates.now()

println("The answer to Problem 1 is: $ans")
println("<< Returned in $(finish - start) milliseconds >>")
